src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy

```diff
@@ -133,22 +143,52 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
                     project.ant.xslt(
                             in: "$xmlReportPath",
                             out: "$htmlReportPath",
-                            style: getToolPath(project, stylesheetRelativePath),
+                            style: getFilePath(project, stylesheetRelativePath),
                             destdir: "${project[toolName].reportsDir}"
                     )
                 }
             }
         }
     }
 
-    private static TextResource getToolResource(Project project, String relativeReference) {
+    private static TextResource getToolResource(Project project, String relativeReference, String overriddenPath) {
         return project.rootProject.resources.text.fromFile(
-                getToolPath(project, relativeReference)
+                getFilePath(project, relativeReference, overriddenPath)
         )
     }
 
-    private static String getToolPath(Project project, String relativeReference) {
-        return "${project.rootDir}/${CODE_QUALITY_DIR}/${relativeReference}"
+    private static TextResource getToolResource(Project project, String relativeReference) {
+        return getToolResource(project, relativeReference, null)
+    }
+
+    private static String getFilePath(Project project, String localPath, String overriddenPath) {
+        if (overriddenPath == null || overriddenPath.isEmpty()) {
+            // Gets file from plugins resources locally and copy to project
+            return getAndCopyFileFromPluginsToProject(project, localPath)
+        }
+        return "${project.rootDir}/${overriddenPath}"
+    }
+
+    private static String getFilePath(Project project, String localPath) {
+        return getFilePath(project, localPath, null)
+    }
+
+
+    /**
+     * Copy file from plugins to root project to the code-quality directory(used for idea plugins).
+     *
+     * @param project from which you run plugins
+     * @param path to file from plugins directories
+     * @return created file path for project
+     */
+    private static String getAndCopyFileFromPluginsToProject(Project project, String path) {
```

Comment:
If it can be static it means isn't modify this object, so if I want this method to be stateless in future, this static restriction is better I think. Idea highlighted this method if it has default modifiers. Also, i don't think private static methods should be named especially, they usually have italic font.